src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -53,427 +41,374 @@ export interface ComboBoxProps {
-    this.selectNextOption();
-    this.handlePopoverOpen;
-  };
+    if (selectedIndex <= 0) {
+      newIndex = navigableOptions.length - 1;
+    } else {
+      newIndex--;
+    }
 
-  private handleUpArrow = () => {
-    this.selectPreviousOption();
-    this.handlePopoverOpen;
-  };
+    selectOptionAtIndex(newIndex);
+  }, [navigableOptions, selectOptionAtIndex, selectedIndex]);
 
-  private handleEnter = (event: KeyboardEvent) => {
-    if (event.keyCode !== Key.Enter) {
-      return;
-    }
+  const selectOptions = useCallback(
+    (selected: string[]) => {
+      selected && onSelect(selected);
+      if (!allowMultiple) {
+        resetVisuallySelectedOptions();
+        setPopoverActive(false);
+        setPopoverWasActive(false);
+      }
+    },
+    [allowMultiple, onSelect, resetVisuallySelectedOptions],
+  );
 
-    const {selectedOption} = this.state;
-    if (this.state.popoverActive && selectedOption) {
-      if (isOption(selectedOption)) {
-        event.preventDefault();
-        this.handleSelection(selectedOption.value);
+  const handleSelection = useCallback(
+    (newSelected: string) => {
+      let newlySelectedOptions = selected;
+      if (selected.includes(newSelected)) {
+        newlySelectedOptions.splice(
+          newlySelectedOptions.indexOf(newSelected),
+          1,
+        );
+      } else if (allowMultiple) {
+        newlySelectedOptions.push(newSelected);
       } else {
-        selectedOption.onAction && selectedOption.onAction();
+        newlySelectedOptions = [newSelected];
       }
-    }
-
-    this.handlePopoverOpen;
-  };
-
-  private handleFocus = () => {
-    this.setState({popoverActive: true, popoverWasActive: true});
-  };
 
-  private handleBlur = () => {
-    this.setState({popoverActive: false, popoverWasActive: false}, () => {
-      this.resetVisuallySelectedOptions();
-    });
-  };
-
-  private handleClick = () => {
-    !this.state.popoverActive && this.setState({popoverActive: true});
-  };
-
-  private handleSelection = (newSelected: string) => {
-    const {selected, allowMultiple} = this.props;
-    let newlySelectedOptions = selected;
-    if (selected.includes(newSelected)) {
-      newlySelectedOptions.splice(newlySelectedOptions.indexOf(newSelected), 1);
-    } else if (allowMultiple) {
-      newlySelectedOptions.push(newSelected);
-    } else {
-      newlySelectedOptions = [newSelected];
-    }
-
-    this.selectOptions(newlySelectedOptions);
-  };
+      selectOptions(newlySelectedOptions);
+    },
+    [allowMultiple, selectOptions, selected],
+  );
 
-  private selectOptions = (selected: string[]) => {
-    const {onSelect, allowMultiple} = this.props;
-    selected && onSelect(selected);
-    if (!allowMultiple) {
-      this.resetVisuallySelectedOptions();
-      this.setState({popoverActive: false, popoverWasActive: false});
-    }
-  };
+  const handleDownArrow = useCallback(() => {
+    selectNextOption();
+    handlePopoverOpen;
```

Comment:
This is just a reference to a function instead of calling the function. So this line does nothing (in the same way a line containing just `"hi"` does nothing)

Seems like this was a bug in the old code too - if it everything worked before then this can be deleted

Reply:
I was wondering about this too 😄